### DISCUSSION TITLE
QTGMC slower than I remembered

### DISCUSSION THREAD

**Discussion by bobdig** [staxrip/staxrip#1442] [General]
I thought that my 16 Core can be utilized at a 100% with QTGMC back in the day with VS. But now it takes ages and my machine is only using two cores fully.

**Comment by Dendraspis** [comment]
AviSynth or VapourSynth?

**Reply by jessielw** [reply]
He said VS so he likely means VapourSynth. To my knowledge QTGMC on VapourSynth should be fully threaded without any additional args. 

However I spit qtgmc ever loaded a 16 core cpu to 100%.

**Reply by bobdig** [reply]
> AviSynth or VapourSynth?

VapourSynth. Haven't tried with AviSynth for a long time. 
QTGMC is still the best deinterlacer as far as I know or is there something better I am not aware of.

**Reply by Dendraspis** [reply]
In that case I would say it depends on your source, encoder settings and QTGMC parameters how much CPU power QTGMC gets or is able to use or simply uses.

**Comment by bobdig** [comment]
I did some testing. The difference between AS and VS is smaller than it used to be, still some benefit for using VS but it is still to slow. 
I then tested with another app called Hybrid. There I get 100% CPU utilization that I used to get from StaxRip some time ago. It even has an GUI-Option for using OpenCL which improves speed a lot but still gives 100% CPU utilization.
`Edit: Tested with 1080i25 content with preset slower and encoder UT Video `
So am I the first to notice or is everyone using something better than QTGMC I do wonder.

**Reply by jessielw** [reply]
Hybrid is likely just automatically enabling the GPU acceleration. Send an example of the output script from hybrid.

**Reply by bobdig** [reply]
I don't know how to do it, is it this file?
[tempPreviewVapoursynthFile15_24_31_806.txt](https://github.com/user-attachments/files/16480721/tempPreviewVapoursynthFile15_24_31_806.txt)

**Reply by jessielw** [reply]
Yeah, he's using GPU acceleration with `TFF=True`. 

If you add this line in staxrip in the qtgmc filter (as long as  the lib is imported, you'll get the same performance)

**Reply by jessielw** [reply]
Update to this, TFF doesn't seem to make much of a difference. There is likely an issue else where. 

Maybe here? 
https://github.com/staxrip/staxrip/issues/1131

**Comment by Selur** [comment]
> Hybrid is likely just automatically enabling the GPU acceleration. Send an example of the output script from hybrid.

Hybrid does not 'automatically enabling the GPU acceleration'.

> Yeah, he's using GPU acceleration with TFF=True.

No. TFF = top field first has nothing to do with GPU acceleration.
When using GPU acceleration with QTGMC in Hybrid the QTGMC call would contain: `opencl=True'
(As a side note, he isn't using GPU decoding either)

> There is likely an issue else where.

Yes, this is more likely the cause of the difference in CPU acceleration.
Then opening Vapoursynth scripts directly with ffmpeg https://forum.videohelp.com/threads/397728-ffmpeg-accepting-vapoursynth-vpy-input-directly-and-gpu-accelerated-speed might be interesting. (not sure whether vapoursynth_alt made it into the official ffmpeg builds)
